### PR TITLE
* Address Issue #156 - Scroll with output

### PR DESCRIPTION
**Pull request by bjaminn**

### CODE REVIEW

**Comment by rgbkrk** [comment]
@ygbr, how does this look?

**Comment by bjaminn** [comment]
I realized the first commit did not limit the scrolling to only when already at the end.  The new commit adds this.

File: lib/script-view.coffee [hunk 1]
```diff
@@ -286,10 +286,20 @@ class ScriptView extends View
...
+      @script.scrollBottom() == (padding + @output.trueHeight())
+
+    lessThanFull = @output.trueHeight() <= @script.trueHeight()
+
     @output.append $$ ->
       @pre class: "line #{css}", =>
         @raw line
 
+    if atom.config.get('script.scrollWithOutput')
+      if lessThanFull || scrolledToEnd
```
Comment (erran) [inline lib/script-view.coffee]: Prefer `or` over `||`.

File: lib/script-view.coffee [hunk 2]
```diff
@@ -286,10 +286,20 @@ class ScriptView extends View
 
     line = @ansiFilter.toHtml(line)
 
+    padding = parseInt(@output.css("padding-bottom"))
```
Comment (erran) [inline lib/script-view.coffee]: Use single quotes unless double quotes are needed for interpolation.

**Comment by erran** [comment]
Running `examples/color.py` I see the output scroll half way through. This script prints all output almost instantly though.

I created a simple script to test and it worked flawlessly scrolling the output. https://gist.github.com/erran/77816419a3710cae6a41

Looks good to me.

**Comment by rgbkrk** [comment]
```
Preparing and tagging a new version ✓
Pushing v2.18.0 tag ✓
Publishing script@v2.18.0 ✓
```

**Comment by JeanMertz** [comment]
@rgbkrk I am seeing `script` not scrolling a `cucumber` feature properly. It seems to scroll one or two lines, but never goes all the way to the bottom.

I couldn't find this mentioned anywhere, is this a know bug? Given @erran's remark:

> Running examples/color.py I see the output scroll half way through. This script prints all output almost instantly though.

**Pull request merged**